src/java.base/share/classes/jdk/internal/foreign/Utils.java

```diff
@@ -96,7 +96,19 @@ static VarHandle put(ValueLayout layout, VarHandle handle) {
                 VarHandle prev = HANDLE_MAP.putIfAbsent(layout, handle);
                 return prev != null ? prev : handle;
             }
+
+            static VarHandle get(ValueLayout layout) {
+                return HANDLE_MAP.get(layout);
+            }
         }
+        layout = layout.withoutName(); // name doesn't matter
+        // keep the addressee layout as it's used below
+
+        VarHandle handle = VarHandleCache.get(layout);
+        if (handle != null) {
+            return handle;
+        }
```

Comment:
Added a cache lookup here so that we don't end up re-creating the var handle if one is already in the cache. Also, I noticed that previously the cache was sensitive to layout names, which it shouldn't I think, since the names are not of consequence to the produced var handle.